src/PowerShellEditorServices/Services/PowerShellContext/Console/WindowsConsoleOperations.cs

```diff
@@ -37,11 +37,7 @@ public async Task<ConsoleKeyInfo> ReadKeyAsync(bool intercept, CancellationToken
             await _readKeyHandle.WaitAsync(cancellationToken).ConfigureAwait(false);
             try
             {
-                return
-                    _bufferedKey.HasValue
-                        ? _bufferedKey.Value
-                        : await Task.Factory.StartNew(
-                            () => (_bufferedKey = System.Console.ReadKey(intercept)).Value);
+                return _bufferedKey ?? await Task.Run(() => (_bufferedKey = System.Console.ReadKey(intercept)).Value).ConfigureAwait(false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Extract the assignment of '_bufferedKey' from this expression.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4918219)